src/cc/usdt/usdt.cc

```diff
@@ -238,12 +238,18 @@ void Context::_each_probe(const char *binpath, const struct bcc_elf_usdt *probe,
 int Context::_each_module(const char *modpath, uint64_t, uint64_t, uint64_t,
                           bool, void *p) {
   Context *ctx = static_cast<Context *>(p);
+
+  std::string path = modpath;
+  if (ctx->pid_) {
+    path = tfm::format("/proc/%d/root%s", *ctx->pid_, path);
+  }
+
   // Modules may be reported multiple times if they contain more than one
   // executable region. We are going to parse the ELF on disk anyway, so we
   // don't need these duplicates.
-  if (ctx->modules_.insert(modpath).second /*inserted new?*/) {
+  if (ctx->modules_.insert(path).second /*inserted new?*/) {
     ProcMountNSGuard g(ctx->mount_ns_instance_.get());
```

Comment:
PID namespaces corresponding to the prefixed paths are now different since you've moved to the process namespace ?

Reply:
I'm not sure I understand this comment. Are saying that this NSGuard will alter the PID namespace of the process running this code to be this PID namespace of the target pid? If so, I don't think this will happen, since we're [`setns`ing with `CLONE_NEWNS` here](https://github.com/iovisor/bcc/blob/master/src/cc/ns_guard.cc#L70). 

That being said, I plan on removing all `ProcMountNSGuard`s from `usdt.cc` as part of this PR, just haven't pushed that up yet.

Reply:
Ah, I see what you mean. Different mount namespace may have different `/proc` mounted. Guess that makes removing NSGuards a necessity.